packages/apollo-cache-control/src/index.ts

```diff
@@ -146,20 +147,32 @@ export const plugin = (
         // own computation onto the `requestContext` for other plugins to read.
```

Comment:
See this comment? I don't know if this concept is documented anywhere else other than in this comment, but I think you can achieve the same results just by writing your own plugin. For example, this is how the response cache plugin works: it reads requestContext.overallCachePolicy in its willSendResponse method. You do need to make sure your plugin runs after the cache control plugin... but that's actually done for you because the cache control plugin is automatically installed before your own plugins.

That is, I think you can achieve your goals with:

```
new ApolloServer({
  cacheControl: {calculateHttpHeaders: false},
  plugins: [{
    requestDidStart() {
      return {
        willSendResponse(requestContext) {
          const {response, overallCachePolicy} = requestContext;
          if (overallCachePolicy && response.http) {
            response.http.headers.set(/* whatever you'd like */
          }
        }
      };
    }
  }],
  /* ... */
})
```

(Code written in GitHub and not tested.)

Do you agree that this would solve your problem?

If so, I think the fix needed is purely documentation. I think you did a great job on the README part of this change; if you want to update this PR to document how to set arbitrary HTTP headers from a plugin in the apollo-cache-control README and/or in `docs/source/performance/caching.md` that would be great!  I'm going to close this PR for now but am happy to reopen it if it turns into a docs PR or if you disagree with my suggestion that this doesn't require an API change.

(Sorry for the delay in review. Apollo Server is now under active maintenance again.)

Reply:
Thanks a lot @glasser this worked for me. Here's an up to date code for anyone who tried this and it failed using apollo 3.7+:
```
const server = new ApolloServer({
  typeDefs,
  resolvers,
  plugins: [
    ApolloServerPluginCacheControl({ calculateHttpHeaders: false }),
    {
      async requestDidStart() {
        return {
          async willSendResponse(requestContext) {
            const { response, overallCachePolicy } = requestContext;
            if (overallCachePolicy && response?.http && overallCachePolicy.maxAge) {
              response.http.headers.set(
                "cache-control",
                `max-age=0, s-maxage=${overallCachePolicy.maxAge}, public` // or any header name you want
              );
            }
          },
        };
      },
    },
  ],
});
```

Reply:
@glasser I made a PR to fix the lack of documentation: https://github.com/apollographql/apollo-server/pull/6415